src/org/omegat/core/data/DataUtils.java

```diff
@@ -25,12 +25,34 @@
 
 package org.omegat.core.data;
 
+import java.util.Set;
+import java.util.TreeSet;
+
 public final class DataUtils {
 
+    private static final Set<String> INTERNAL_IDS = new TreeSet<>();
```

Comment:
Now this is alive during OmegaT application run.
This should be a project unique so it should be in ProjectTMX or RealProject class.